first_commit

AIeksandrKotov opened the pull request.

AIeksandrKotov committed:
first_commit

pomegranate282922 commented on src/main/java/Page/LoginPage.java:
```diff
@@ -0,0 +1,145 @@
+package Page;
```
Пакеты именуем строчными буквами без camelCase.
Нужно исправить

pomegranate282922 commented on src/main/java/Page/MainPageQuestions.java:
```diff
@@ -0,0 +1,62 @@
+    // Куки
+    private final By cookie = By.xpath("//button[@class='App_CookieButton__3cvqF']");
+    // Вопросы
+    public static  final By oneQuestions = By.id("accordion__heading-0");
```
Зачем статик?
Нужно сделать однообразно во всех po: или все элементы-поля статик или нет. 
static final - это константы, именуем заглавными буквами через нижнее подчеркивание 
Нужно исправить

AIeksandrKotov replied:
Исправлено

pomegranate282922 commented on src/main/java/Page/MainPageQuestions.java:
```diff
@@ -0,0 +1,62 @@
+    public static  final By fiveResponse = By.id("accordion__panel-4");
+    public static  final By sixResponse = By.id("accordion__panel-5");
+    public static  final By sevenResponse = By.id("accordion__panel-6");
+    public static  final By eightResponse = By.id("accordion__panel-7");
```
Все элементы должны быть приватными чтобы соблюсти принцип инкапсуляции 
Нужно исправить

AIeksandrKotov replied:
Сделал все поля приватными

pomegranate282922 commented on src/test/java/PageTest/BaseTest.java:
```diff
@@ -0,0 +1,39 @@
+import java.util.concurrent.TimeUnit;
+
+public class BaseTest {
+    public static final String mainPage = "https://qa-scooter.praktikum-services.ru";
```
Именуем как константы 
Нужно исправить

AIeksandrKotov replied:
Исправлено

pomegranate282922 commented on src/test/java/PageTest/BaseTest.java:
```diff
@@ -0,0 +1,39 @@
+    public WebDriver driver;
+
+
+    @Before
```
Отлично что сделан общий предок

pomegranate282922 commented on src/test/java/PageTest/TestPageLogin.java:
```diff
@@ -0,0 +1,57 @@
+package PageTest;
```
Нейминг пакетов нужно подправить

pomegranate282922 commented on src/test/java/PageTest/TestPageLogin.java:
```diff
@@ -0,0 +1,57 @@
+    private final String color;
+    private final String comment;
+    // Создаем локатор для поиска элемента
+    public static final By orderPlaced =
```
Локаторов в тестах быть не должно. Это нарушает Page object model.
Нужно исправить

AIeksandrKotov replied:
Убрал локатор из теста

pomegranate282922 commented on src/main/java/Page/LoginPage.java:
```diff
@@ -0,0 +1,145 @@
+    }
+
+    // Метод авторизации в приложении
+    public void Login(String username, String firstName, String address, int metroName, String numberPhone, String dateOrder, int leasPeriod, String color, String comment){
```
Все методы именуем начиная со строчной буквы 
Нужно исправить

AIeksandrKotov replied:
Исправлено

pomegranate282922 commented on src/test/java/PageTest/TestPageLogin.java:
```diff
@@ -0,0 +1,57 @@
+                .Login(userName, firstName, address, metroName, numberPhone, dateOrder, leasPeriod, color, comment);
+        String text = driver.findElement(orderPlaced).getText();
+        MatcherAssert.assertThat("Заказ не оформлен",text, containsString("Заказ оформлен"));
+    }
```
Надо использовать минимум два тестовых набора 
Нужно исправить

pomegranate282922 replied:
Нет закрытия браузера

pomegranate282922 replied:
Логика должна быть такая:
1. Флоу с одной кнопкой протестирован до конца двумя разными тестовыми данными
2. Второй флоу ( нижняя кнопка) - проверить, что открывается окно для ввода параметров заказа 
Нужно исправить

AIeksandrKotov replied:
Добавил вторую кнопку и тест по ней, добавил закрытие браузера

pomegranate282922 commented on src/test/java/PageTest/TextQuestions.java:
```diff
@@ -0,0 +1,61 @@
+// Создаем поля где будут храниться параметры тестового метода
+@RunWith(Parameterized.class)
+public class TextQuestions extends BaseTest {
+    private final By Questions;
```
Нейминг нужно исправить

AIeksandrKotov replied:
Исправил везде нейминг

pomegranate282922 commented on src/test/java/PageTest/TextQuestions.java:
```diff
@@ -0,0 +1,61 @@
+
+
+    // принимаем значение полей
+    public TextQuestions(By questionsItem, By responsesItem, String TextResponse) {
```
Отлично, что применена параметризация

pomegranate282922 reviewed (commented):
Хорошая работа! Реализована POM, применена параметризация. 
есть и замечания:
1. Нейминг методов, переменных, пакетов 
2. Все po должны быть сделаны единообразно 
3. Поля в po приватные 
4. В тестах не должно быть локаторов 
5. В тестах на заказ сделать два тестовых набора и тест на вторую кнопку 
6. Остальное по мелочи

AIeksandrKotov committed:
Исправлено - Нейминг методов, переменных, пакетов
Исправлено - Поля в po приватные
Исправлено - В тестах не должно быть локаторов
Исправлено - В тестах на заказ сделать два тестовых набора и тест на вторую кнопку
Исправлено - Остальное по мелочи

AIeksandrKotov committed:
Исправлено - Нейминг методов, переменных, пакетов
Исправлено - Поля в po приватные
Исправлено - В тестах не должно быть локаторов
Исправлено - В тестах на заказ сделать два тестовых набора и тест на вторую кнопку
Исправлено - Остальное по мелочи

AlexanderKotov89 commented:
+ исправил проблему с chromedriver, добавил в BaseTest запуск браузера сhrome

pomegranate282922 commented on src/main/java/page/LoginPage.java:
```diff
@@ -0,0 +1,167 @@
+    }
+
+    private final By COOKIE = By.xpath("//button[@class='App_CookieButton__3cvqF']"); // Куки
+    private final By BUTTON_ORDER_TOP = By.xpath("//button[@class='Button_Button__ra12g']"); // Верхняя кнопка заказать
```
Так: нестатические переменные именуем начиная со строчной буквы camelCase.
Нужно исправить везде

pomegranate282922 commented on src/main/java/page/LoginPage.java:
```diff
@@ -0,0 +1,167 @@
+        setOrderColor(color);
+        setOrderComment(comment);
+        setButtonRequest();
+        setButtonConfirm();
```
Получилось два метода почти одинаковых. Проще вынести отличающийся метод отдельно. А еще лучше сделать, чтобы через параметр определялось какую из кнопок заказать нажимать.
Можно улучшить

pomegranate282922 commented on src/test/java/pagetest/TestPageLogin.java:
```diff
@@ -0,0 +1,68 @@
+                .loginButtonLower(USER_NAME, FIRST_NAME, ADDRESS, METRO_NAME, NUMBER_PHONE, DATA_ORDER, LEAS_PERIOD, COLOR, COMMENT);
+        String text = driver.findElement(ORDER_PLACED).getText();
+        MatcherAssert.assertThat("Заказ не оформлен",text, containsString("Заказ оформлен"));
+    }
```
Немного другое ожидаем.
Нужно:
1. Первый флоу протестирован до конца с 2мя наборами тестовых данных ( это есть)
2. Второй флоу ( нижняя кнопка) проверить, что происходит выход на создание заказа. 

Нужно исправить

pomegranate282922 replied:
Немного все же не доделал логику, как ожидалось. У тебя получилось два одинаковых теста на оба флоу. А здорово было бы второй флоу закончить на моменте выхода на форму ввода данных заказа, чтобы не дублировать. 
можно улучшить

pomegranate282922 reviewed (commented):
Многое исправил, супер!
Осталось 2 замечания
1. С неймингом разобраться 
2. В тестах на создание заказа немного переделать

AIeksandrKotov committed:
В LoginPage сделал все поля static что бы везде все выглядело одинаково, поправил нейминг в переменных
Добавил в метод Login выбор кнопки, теперь в тестовых данных есть кнопки которые оформляют заказ

AlexanderKotov89 commented:
С неймингом вроде разобрался. Названия методов, название полей классов написал со строчной буквы, только константы static final написал заглавными буквами через нижнее подчеркивание. Название классов написал с прописной буквы

AlexanderKotov89 commented:
А вот с кнопкой пришлось повозиться, выносить отдельный метод я не хотел что-бы не плодить зло, решил улучшить, передал в параметр кнопку верхнюю и нижнюю, вроде выглядит не плохо)
Пришлось покумекать как это сделать ))

pomegranate282922 commented on src/main/java/page/LoginPage.java:
```diff
@@ -0,0 +1,155 @@
+    // Метод авторизации в приложении
+    public void login(String button, String userName, String firstName, String address, int metroName, String numberPhone, String dateOrder, int leasPeriod, String color, String comment){
+        clickCookie();
+        if (button.equals("topButton")) {
```
Умница!

pomegranate282922 reviewed (approved):
Критичных замечаний нет - зачет!
